File: Day251/findTheClosestPairFromTwoArrays.java
package Day251;

import java.util.ArrayList;
import java.util.List;

public class findTheClosestPairFromTwoArrays {

    public static List<Integer> printClosest(int arr[], int brr[], int x) {
        List<Integer> answer = new ArrayList<>();
        
         int min = Integer.MAX_VALUE;
         int i =0, j = brr.length-1;

         while (i<arr.length && j >= 0) {
            int temp = arr[i] + brr[j];
            int difference = Math.abs(temp - x);
            if (difference < min) {
                min = difference;
                if (answer.size() == 0) {
                    answer.add(arr[i]);
                    answer.add(brr[j]);
                } else {
                    answer.set(0, arr[i]);
                    answer.set(1, brr[j]);
                }
            }
            if (temp < x) {
                i++;
            }else {
                j--;
            }
         }

        
        return answer;
    }

    public static void main(String[] args) {
        int arr[] = {1, 4, 5, 7};
        int brr[] = {10, 20, 30, 40};
        int x = 32;
        List<Integer> answer = printClosest(arr, brr, x);
        System.out.println(Math.abs(answer.get(0)+answer.get(1)- x));
    }
}

File: Day251/findTheClosestPairFromTwoArrays.go
package main

import (
	"fmt"
	"math"
)

func printClosest(arr []int, brr []int,x int ) []int {
	answer := make([]int ,2)
	i, j := 0, len(brr) -1
	min := math.MaxInt
	for i < len(arr) && j >= 0 {
		temp := arr[i] + brr[j]
		diff := math.Abs(float64(temp - x))
		if int(diff) < min {
			min = int(diff)
			answer[0] = arr[i]
			answer[1] = brr[j]
		}
		if temp < x {
			i++
		} else {
			j--
		}
	}
	return answer
}


func main() {
	arr := []int{1, 4, 5, 7}
	brr := []int{10, 20, 30, 40}
    x := 32
	answer := printClosest(arr, brr, x)
	fmt.Println(math.Abs(float64(answer[0]+answer[1]-x)))
}